src/libDirectoryService/DSBlockPostProcessing.cpp

```diff
@@ -233,38 +230,47 @@ void DirectoryService::UpdateMyDSModeAndConsensusId() {
                                             .GetHashForRandom()
                                             .asBytes());
   }
-  // Check if I am the oldest backup DS (I will no longer be part of the DS
-  // committee)
-  if ((uint32_t)(m_consensusMyID + numOfIncomingDs) >=
-      m_mediator.m_DSCommittee->size()) {
+
+  // Find my new consensus ID.
+  bool isDropout = true;
```

Comment:
To check for Dropout, I think now we can make use of `m_removeDSNodePubkeys`

Reply:
`dropout` is on two possibilities:

1. is a removed node
2. is an expired node.

This covers both possibilities so I think handling them separately is not the best.

Reply:
To clarify, the `m_removeDSNodePubkeys` field contains only removals for non-performance as suggested in the @ShengguangXiao's proposal to add the new field. The expired nodes are calculated in `UpdateDSCommitteeComposition`. I think this is better than putting expired nodes in the field because it allows some distinction deterministically when just looking at the DS block alone.

Reply:
I see. Then it does make sense. Maybe we can add an inline comment for the possibilities of dropout
```
is a removed node
is an expired node
```

Reply:
Thanks for the suggestion. Included more comments to make it clearer.